docs/architecture/adr-017-single-binary-upgrade.md

```diff
@@ -0,0 +1,141 @@
+# ADR 017: Single binary upgrade
+
+## Status
+
+Proposed
+
+## Changelog
+
+- 2023/03/22: initial draft
+
+## Context
+
+It important to sync from genesis with one binary.
+It is needed for wallets, block explorers, indexers, analytics (e.g. Nansen).
+It makes it possible to sync headers from chain and download blobs from third party service.
+
+## Alternative Approaches
+
+Instead of having a single binary, we can have multiple binaries that are managed by another supervisory binary e.g. [Cosmovisor](https://docs.desmos.network/fullnode/cosmovisor/).
+Osmosis does it through having a significant number of binaries gathered in one docker container and switch them in particular heights.
+This approach has a lot of maintenance burden and is fragile to sync from genesis.
```

Comment:
> This approach has a lot of maintenance burden

I'm not convinced that the Osmosis approach has any more maintenance burden than what we're proposing with this ADR.

> and is fragile to sync from genesis

Are there scenarios where Osmosis sync from genesis fails? Should we include data to support this point?